instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/ClientSideServerPortCondition.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.http;
+
+import java.util.function.BiPredicate;
+
+final class ClientSideServerPortCondition<REQUEST> implements BiPredicate<Integer, REQUEST> {
+
+  private final HttpClientAttributesGetter<REQUEST, ?> getter;
+
+  ClientSideServerPortCondition(HttpClientAttributesGetter<REQUEST, ?> getter) {
+    this.getter = getter;
+  }
+
+  @Override
+  public boolean test(Integer port, REQUEST request) {
+    String url = getter.getUrlFull(request);
+    if (url == null) {
+      return true;
+    }
+    // according to spec: extract if not default (80 for http scheme, 443 for https).
+    if ((url.startsWith("http://") && port == 80) || (url.startsWith("https://") && port == 443)) {
```

Comment:
Should we lowercase in the event that some implementations return the original url, which could possibly have upper or mixed case? Lowercase is canonical, but if a URL somehow was entered uppercase AND the Getter implementation passed it through, it would fail this.

Reply:
Good point 👍